src/opds_dumper.cpp

```diff
@@ -242,10 +237,27 @@ std::string OPDSDumper::categoriesOPDSFeed() const
   );
 }
 
+void fillLanguagesMap() 
+{
+  auto isoLanguages = icu::Locale::getISOLanguages();
+  while (*isoLanguages != NULL) {
+    auto lang = *isoLanguages;
+    const icu::Locale locale(lang);
+    icu::UnicodeString ustring;
+    locale.getDisplayLanguage(locale, ustring);
+    std::string result;
+    ustring.toUTF8String(result);
+    std::string iso3LangCode = locale.getISO3Language();
+    iso639_3.insert({iso3LangCode, result});
+    ++isoLanguages;
+  }
+}
+
 std::string OPDSDumper::languagesOPDSFeed() const
 {
   const auto now = gen_date_str();
   kainjow::mustache::list languageData;
+  fillLanguagesMap();
```

Comment:
Does it make sense to have this iso639_3 map be available by a method in kiwix/tools.h?
That way it can be used by Kiwix-Desktop also (helping in this: https://github.com/kiwix/kiwix-desktop/issues/681)

Reply:
This is a tricky question... If I understand properly what implies your question is: Should Kiwix Desktop use the REST API or directly use the libkiwix primitive to populate its own language list? I'm in favour of the second for performance/pragmatical reasons. The advantages of having the language list pulled from the online REST API at https://library.kiwix.org/ seems not big enough to me. But interested in @veloman-yunkan's opinion on this.

Reply:
As long as the language mapping API is well thought out I don't have any argument against exposing it via `kiwix/tools.h`. However I am not sure why you may need it in kiwix-desktop. If you are trying to close with it a gap present in the libkiwix API, maybe it can be fixed in a better way.